spec/takeaway_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'takeaway'
+
+describe Takeaway do
+
+  subject(:takeaway) { described_class.new(menu) }
+  let(:menu) { double(:menu, dishes: { "pepperoni" => 7.50, "hawaiian" => 6.50, "vegetarian" => 5.50 }) }
+
+# think about extracting Text/sms to its own class
+#  let(:sms) { double(:sms, deliver_sms: nil) }
+
+
+  describe "#view_menu" do
+    it "shows a shows a list of dishes with prices" do
+      allow(menu).to receive(:print_menu).and_return("Pepperoni £7.50")
+      expect(takeaway.view_menu).to eq("Pepperoni £7.50")
+    end
```

Comment:
I would argue that this test goes a little deeper than it needs to.

Your view_menu method looks like this:
```ruby
def view_menu
  menu.print_menu
end
```

It's only responsibility is to send a print_menu message to whatever is held in the @menu instance variable.  It doesn't know or care what the will happen next - nor should it.

A better test here would be something like:
```ruby
it 'tells the menu to print' do
  expect(menu).to receive(:print_menu)
  takeaway.view_menu
end
```

Testing the actual output of doing this belongs in a test for print_menu within the Menu class.